pkg/kubelet/nodestatus/setters.go

```diff
@@ -249,6 +279,76 @@ func hasAddressValue(addresses []v1.NodeAddress, addressValue string) bool {
 	return false
 }
 
+func findFqdn(name string, domains []string) string {
+	if len(domains) < 1 {
+		return name
+	}
+
+	// If our hostname already ends with one of the search domains, treat it as a FQDN
+	for _, d := range domains {
+		if strings.HasSuffix(name, d) {
+			return name
+		}
+	}
+
+	// Otherwise, use the first search domain as the conventional DNS domain
+	return name + "." + domains[0]
+}
+
+func findHostIPsViaDNS(name string, preferIPv4 bool) (out []net.IP) {
```

Comment:
golang has already 2 resolvers https://golang.org/pkg/net/#hdr-Name_Resolution, very personal opinion, I'm afraid adding a 3rd one can makes things more difficult

Reply:
Neither of the internal resolvers is capable of ignoring `/etc/hosts` entries.  Again, that's normally fine and preferred; just not in this case.